### ISSUE TITLE
Error executing: "C:\Program Files\TortoiseHg\hg.exe" in -q default

### ISSUE THREAD

**Issue by overlord** [wbond/package_control#1025]
```bash
Package Control: Fetching list of available packages
  Platform: windows-x64
  Sublime Text Version: 2221
  Package Control Version: 3.1.2
Package Control: Looking for git at: <blah-blah-blah>
Package Control: Found git at "C:\Program Files (x86)\Git\bin\git.exe"
Package Control: Executing "C:\Program Files (x86)\Git\bin\git.exe" symbolic-ref -q HEAD [C:\Portable\Sublime Text 2\Data\Packages\Markdown Preview]
Package Control: Executing "C:\Program Files (x86)\Git\bin\git.exe" config --get branch.master.remote [C:\Portable\Sublime Text 2\Data\Packages\Markdown Preview]
Package Control: Executing "C:\Program Files (x86)\Git\bin\git.exe" config --get branch.master.merge [C:\Portable\Sublime Text 2\Data\Packages\Markdown Preview]
Package Control: Executing "C:\Program Files (x86)\Git\bin\git.exe" fetch origin [C:\Portable\Sublime Text 2\Data\Packages\Markdown Preview]
Package Control: WinINet HTTPS Debug General
  Closing connection to packagecontrol.io on port 443 after 1 request
Package Control: Executing "C:\Program Files (x86)\Git\bin\git.exe" log ..origin/master [C:\Portable\Sublime Text 2\Data\Packages\Markdown Preview]
Package Control: Looking for hg at: <blah-blah-blah>
Package Control: Found hg at "C:\Program Files\TortoiseHg\hg.exe"
Package Control: Executing "C:\Program Files\TortoiseHg\hg.exe" in -q default [C:\Portable\Sublime Text 2\Data\Packages\Sublime TFS]
error: Package Control

Error executing: "C:\Program Files\TortoiseHg\hg.exe" in -q default
Exit code: 1

VCS-based packages can be ignored with the "ignore_vcs_packages" setting.`
```

As far as I Gooooogled - Mercurial `exit code = 1` is not an error - it's just `nothing to pull` sign.
<https://bugzilla.mozilla.org/show_bug.cgi?id=723638>

And actually they're not kidding :) it can be shown by hg `-v` option:

```bash
C:\Portable\Sublime Text 2\Data\Packages\Sublime TFS>dir
<blah-blah-blah>
C:\Portable\Sublime Text 2\Data\Packages\Sublime TFS>echo %errorlevel%
0
C:\Portable\Sublime Text 2\Data\Packages\Sublime TFS>"C:\Program Files\TortoiseHg\hg.exe" in -q default -v
comparing with https://bitbucket.org/CDuke/sublime-tfs
searching for changes
no changes found
C:\Portable\Sublime Text 2\Data\Packages\Sublime TFS>echo %errorlevel%
1
```

So maybe it's a good idea to check returncode in `C:\Portable\Sublime Text 2\Data\Packages\Package Control\package_control\cmd.py` like this?

```python
<...>
output = output.decode(encoding)
output = output.replace('\r\n', '\n').rstrip(' \n\r')

binary_name = os.path.basename(args[0])
if re.search('hg', binary_name):
    valid_returncodes = [0, 1]
else:
    valid_returncodes = [0]

if proc.returncode not in valid_returncodes:
    if not ignore_errors or re.search(ignore_errors, output) is None:
        show_error(
            u'''
            Error executing: %s
            Exit code: %i
            %s
            VCS-based packages can be ignored with the
            "ignore_vcs_packages" setting.
            ''',
            (create_cmd(args), proc.returncode, output)
        )
        return False
<...>
```

**Comment by amclain** [comment]
+1
I'm a Mercurial user and have been dealing with the same error.

**Comment by wbond** [comment]
This should be fixed by f56d383495fff96511ffbff37a21baff05774c0c
